test/c/derive.c

```diff
@@ -163,6 +163,27 @@ void derive_single_line_with_custom_knots(CuTest *tc)
 		TS_CALL(try, status.code, ts_bspline_set_knot_at(
 			&spline, 3, 2.f, &status))
 
+		/* Confirm the spline is a line with the desired slope. */
+		size_t nsamples = 3;
+		tsReal min, max;
+		ts_bspline_domain(&spline, &min, &max);
+		tsReal span = max - min;
+		tsReal slope[2];
+		slope[0] = (ctrlp[2] - ctrlp[0]) / span;
+		slope[1] = (ctrlp[3] - ctrlp[1]) / span;
+		tsReal step = span / (nsamples - 1);
+		for(size_t i = 0; i < nsamples; ++i) {
```

Comment:
Did you know that TinySpline offers a function to sample splines: `ts_bspline_sample`. It might be easier to use.

Reply:
Great point - I completely forgot about that functionality even though I use it frequently

Reply:
That's a good point.